src/utils.jl

```diff
@@ -57,6 +57,22 @@ function insorted(item, collection; rev=false)
     return !isempty(index)
 end
 
+@inline function insorted(x::T, A::AbstractVector{T}) where T
+    n = length(A)
+    lo = 0
+    hi = n+1
+    @inbounds while hi-lo > 1
+        m = lo + ((hi - lo) >>> 0x01)
```

Comment:
I wonder if using `/ 2` would not also create the same code. Or maybe that would only work for `UInt` ?

Reply:
I simply copy-pasted the `searchsortedfirst` (except that we return a bool instead of the index) code from Julia's standard library in this function. I don't know why it is faster this way, but I assume they would've already optimized it. Although branchless version maybe worth trying, I'll try and see if we can get more speedup from it.

Reply:
Weird, one would have expected that Julia inlines that function and then would generate the same method. Are you using the `-O3` flag when benchmarking?

Reply:
I'm getting pretty much the same results when I started julia with -O3 flag.

Reply:
Also here is the almost-branchless version -
```julia
@inline function insorted(x::T, A::AbstractVector{T}) where T
    n = length(A)
    n == 0 && return false
    half = n >>> 0x01
    lo = 1
    @inbounds while half > 0
       mid = lo + half
       lo = A[mid] <= x ? mid : lo
       n -= half
       half = n >>> 0x01
    end
    return A[lo] == x
end
```
I'm not getting any speedup using this, but maybe there is some optimization I'm missing.

Reply:
I think the tenary operator is not branchless though, for that you have to  use the `ifelse` function.